xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/internal/document/DefaultDocumentAuthors.java

```diff
@@ -0,0 +1,144 @@
+     */
+    public DefaultDocumentAuthors()
+    {
+    }
+
+    /**
+     * Default constructor cloning the given document authors.
+     *
+     * @param documentAuthors the authors information to be cloned.
+     */
+    public DefaultDocumentAuthors(DocumentAuthors documentAuthors)
+    {
+        this.contentAuthor = documentAuthors.getContentAuthor();
+        this.metadataAuthor = documentAuthors.getMetadataAuthor();
+        this.displayedAuthor = documentAuthors.getDisplayedAuthor();
+    }
+
+    /**
+     * Specify the author of the content of the document: this author is only responsible to the content, and not to
+     * other information of the document such as xobjects.
+     *
+     * @param contentAuthor the author of the content of the document.
+     * @return the current instance for Builder pattern.
+     */
+    public DefaultDocumentAuthors setContentAuthor(UserReference contentAuthor)
+    {
+        this.contentAuthor = contentAuthor;
+        return this;
+    }
+
+    /**
+     * Specify the metddata author of the document: this author is not responsible to the content, but responsible to
+     * the xobjects and other metadata.
+     *
+     * @param metdataAuthor the author of the metadata of the document.
+     * @return the current instance for builder pattern.
+     */
+    public DefaultDocumentAuthors setMetadataAuthor(UserReference metdataAuthor)
+    {
+        this.metadataAuthor = metdataAuthor;
+        return this;
+    }
+
+    /**
+     * Specify the displayed author of the document: this author is only there for display purpose, and might be
+     * different from other authors if the document has been saved through a script for example.
+     *
+     * @param displayedAuthor the author to display.
+     * @return the current instance for builder pattern.
+     */
+    public DefaultDocumentAuthors setDisplayedAuthor(UserReference displayedAuthor)
+    {
+        this.displayedAuthor = displayedAuthor;
+        return this;
+    }
+
+    @Override
+    public UserReference getContentAuthor()
+    {
+        return this.contentAuthor;
+    }
+
+    @Override
+    public UserReference getMetadataAuthor()
+    {
+        return this.metadataAuthor;
+    }
+
+    @Override
+    public UserReference getDisplayedAuthor()
+    {
+        return this.displayedAuthor;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        DefaultDocumentAuthors that = (DefaultDocumentAuthors) o;
+
+        return new EqualsBuilder()
+            .append(contentAuthor, that.contentAuthor)
+            .append(metadataAuthor, that.metadataAuthor)
+            .append(displayedAuthor, that.displayedAuthor)
+            .isEquals();
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return new HashCodeBuilder(17, 67)
```

Comment:
What is the passing custom values to this constructor ?

Reply:
The javadoc suggest to pick randomly chosen hardcoded value for those numbers, if possible different for each classes, so that's what I did.

Reply:
That would suggest that two object of incompatible type would be used as key in the same Map, quite strange.